transcendence-app/src/chat/chat.service.ts

```diff
@@ -67,20 +71,27 @@ export class ChatService {
     });
   }
 
-  private async addChatroom(
-    chatroom: Partial<Chatroom>,
+  private async addAvatarAndChatroom(
+    avatarId: string,
+    avatarDto: LocalFileDto,
+    chatroom: ChatroomDto,
   ): Promise<Chatroom | null> {
-    return this.chatroomRepository.addChatroom({
-      id: uuidv4(),
-      createdAt: new Date(Date.now()),
-      avatarX: 0,
-      avatarY: 0,
-      ...chatroom,
-    });
+    return this.chatroomRepository.addAvatarAndAddChatroom(
+      { id: avatarId, createdAt: new Date(Date.now()), ...avatarDto },
+      {
+        id: uuidv4(),
+        createdAt: new Date(Date.now()),
+        avatarX: 0,
+        avatarY: 0,
+        ...chatroom,
+      },
+    );
   }
 
   async createChatroom(ownerId: string, chatroom: CreateChatroomDto) {
     const { confirmationPassword: _, ...newChat } = chatroom;
```

Comment:
If you don't mind, could you refer to me to some source where I can get some information on how does this _ (underscore) syntax work? Thanks

Reply:
If I'm not mistaken, this is a [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment), and we assign the confirmation password to the `_` variable to signal we don't use it.